.github/workflows/try.yml

```diff
@@ -87,7 +90,7 @@ jobs:
     if: ${{ always() && fromJson(needs.parse-comment.outputs.configuration).try}}
     steps:
       - name: Success
-        if: ${{ !contains(join(needs.*.result, ','), 'failure') }}
+        if: ${{ !contains(join(needs.*.result, ','), 'failure') && !contains(join(needs.*.result, ','), 'cancelled') }}
```

Comment:
[`contains`](https://docs.github.com/en/actions/learn-github-actions/expressions#contains) seems to work with arrays directly. Do we still need the `join` here to convert results to a single string?

Reply:
Presumably the array may contain a string that contains `cancelled` but is longer than that?

Reply:
Huh. This is a good question. I would like to try calling join on the array directly, but I'm copying the same approach we use on other workflows here. Maybe I can try adjusting all workflows to use join on the array directly in a follow up change so that it's the same everywhere.

Reply:
Ah, good point, but as Martin says it may be better to adjust them all in a follow-up

Reply:
> Presumably the array may contain a string that contains `cancelled` but is longer than that?

I think isn't expected to happen but maybe we can do this cleanup in all workflows in a follow up:  https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context

Reply:
Agreed, we can have a separate PR for this cleanup.